packages/react-instantsearch/src/widgets/SearchBox.tsx

```diff
@@ -44,10 +51,10 @@ export function SearchBox({
   const [inputValue, setInputValue] = useState(query);
   const inputRef = useRef<HTMLInputElement>(null);
 
-  function setQuery(newQuery: string, compositionComplete = true) {
+  function setQuery(newQuery: string, isComposing = false) {
     setInputValue(newQuery);
 
-    if (searchAsYouType && compositionComplete) {
+    if (searchAsYouType && !(ignoreCompositionEvents && isComposing)) {
```

Comment:
Won't it actually refine when `ignoreCompositionEvents` is `false` here, no matter what the value of `isComposing` is ?
I feel like this is reversed ?

Reply:
I feel like this should be `searchAsYouType && (ignoreCompositionEvents || !isComposing)`